src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_IEqualityComparer.cs

```diff
@@ -11,8 +12,18 @@ internal abstract partial class AbstractAsynchronousTaggerProvider<TTag>
 {
     private partial class TagSource : IEqualityComparer<ITagSpan<TTag>>
     {
+        private readonly ObjectPool<HashSet<ITagSpan<TTag>>> _tagSpanSetPool;
+
         public bool Equals(ITagSpan<TTag>? x, ITagSpan<TTag>? y)
-            => x != null && y != null && x.Span == y.Span && _dataSource.TagEquals(x.Tag, y.Tag);
+        {
+            if (x == y)
+                return true;
```

Comment:
this clause was added as a fast path.  The rest was just broken into separate clauses.